### PR TITLE
Update getting_started.md

### PR DESCRIPTION
**Pull request by mikedropmedia**

### CODE REVIEW

**Comment by cheng-tan** [comment]
Comments in #95 needs to be addressed with PR.

File: _guides/getting_started.md [hunk 1]
```diff
@@ -10,31 +10,60 @@ body_class: tutorial
 resource_icon: /svg/resources/guide.svg
 ---
 
-# A Step by Step Introduction to Linear Regression
+# Getting Started with Linear Regression and Vowpal Wabbit
 
-## A first data-set
+This guide describes how to run Linear Regression with Vowpal Wabbit (VW). It features an overview of a simple regression problem using a VW workflow tutorial with examples. Find information about interacting with VW, including structuring input, understanding output, and learn important VW diagnostics.
```
Comment (mikedropmedia) [inline _guides/getting_started.md]: @kumpera comments: 
Linear Regression is a form of regression, not an algorithm per-se.
Creation of models is one of the many tasks the reader can do, it's not an end goal on itself.
I think "in VW" sounds worse than "with VW" as the result of such regression outlives the usage VW itself.

See updated. Will apply to CB guide as well.

File: _guides/getting_started.md [hunk 2]
```diff
@@ -10,31 +10,60 @@ body_class: tutorial
...
 
-## A first data-set
+This guide describes how to run Linear Regression with Vowpal Wabbit (VW). It features an overview of a simple regression problem using a VW workflow tutorial with examples. Find information about interacting with VW, including structuring input, understanding output, and learn important VW diagnostics. 
 
-Now, let's create a data-set. Suppose we want to predict whether a house will require a new roof in the next 10 years. We can create a training-set file, `house_dataset` with the following contents:
+## Getting started with Vowpal Wabbit
+
+To install VW—and for more information on building VW from source or using a package manager—see [Getting started](https://github.com/VowpalWabbit/vowpal_wabbit/wiki/Getting-started) guide. 
+
+>**Note:** The CB tutorial uses [Vowpal Wabbit Python package](https://github.com/VowpalWabbit/vowpal_wabbit/tree/master/python). Additional binary packages are available for select platforms. See [Getting started](https://cheng-tan.github.io/vowpalwabbit.github.io/) module on the homepage for more information.
```
Comment (mikedropmedia) [inline _guides/getting_started.md]: @kumpera comments: 
supposed is ambiguous as it might mean that it's available from X but not supported.
Support is a loaded technical term that is usually better replaced.
We want to mention Python, we want to highlight Python.

Rewrite to accomodate, will include in CB guide note as well.

File: _guides/getting_started.md [hunk 3]
```diff
@@ -10,31 +10,60 @@ body_class: tutorial
...
+The label information for the second line is more complex than the first: 
 
-The 3rd example is straightforward, except there is an additional number: `0.5` following the importance weight, in the label information. This is an initial prediction. Sometimes you have multiple interacting learning systems and want to be able to predict an offset rather than an absolute value.
+`1 2 'second_house | price:.18 sqft:.15 age:.35 1976`
+
+- The `1` is the label indicating that a roof replacement is required. 
+- The `2` is an optional _importance weight_, which implies that this example counts twice. The importance weight comes up in many settings. 
+- A missing importance weight defaults to 1. `'second_house` is the tag. You can find more information on importance weight in the the Vowpal Wabbit diagnostic header section below.
```
Comment (mikedropmedia) [inline _guides/getting_started.md]: @kumpera comments:
I just realized that the wording of it is used elsewhere to identify the example is quite vague. It does show up later in this tutorial so shouldn't we setup the right expectation?

Does this work Rodrigo? 

@cheng-tan can we add an anchor link to section specified?

File: _guides/getting_started.md [hunk 4]
```diff
@@ -57,64 +86,140 @@ best constant = 0.500000
...
-There is a burble of diagnostic information which you can turn off with `--quiet`. However, it's worthwhile to first understand it, so let's work through it bit by bit.
+## Vowpal Wabbit diagnostics
+
+This section provides information on the various types of diagnostic information VW presents.
+
+**Use** `--quiet`  command to turn off diagnostic information in Vowpal Wabbit.
+
+### Hash function bits
+
+The following VW output shows the number of bits from the hash function:
```
Comment (mikedropmedia) [inline _guides/getting_started.md]: @ kumpera comment:
This is not a code sample, it's from the VW output

I updated all of these references, copy/paste error...

File: _guides/getting_started.md [hunk 5]
```diff
@@ -57,64 +86,140 @@ best constant = 0.500000
...
 
 ```
 Num weight bits = 18
 ```
-Only 18 bits of the hash function will be used. That's much more than necessary for this example. You could adjust the number of bits using `-b bits`
+
+This diagnostic shows that the number of bits from the hash function is 18 (more than enough for this example). Use `-b bits` to adjust the number of bits for a hash function.
```
Comment (mikedropmedia) [inline _guides/getting_started.md]: @kumpera comments:
We use the bits that comes out of the hash function.

Changed

File: _guides/getting_started.md [hunk 6]
```diff
@@ -57,64 +86,140 @@ best constant = 0.500000
...
+### Learning rate decay power
+
+The following VW output shows the power of learning rate decay:
 
 ```
 power_t = 0.5
 ```
-This specifies the power on the learning rate decay. You can adjust this `--power_t p` where _p_ is in the range [0,1]. 0 means the learning rate does not decay, which can be helpful when state tracking, while 1 is very aggressive, but plausibly optimal for [IID](https://en.wikipedia.org/wiki/Independent_and_identically_distributed_random_variables) data-sets. 0.5 is a minimax optimal choice. A different way of stating this is: stationary data-sets where the fundamental relation between the input features and target label are not changing over time, should benefit from a high (close to 1.0) `--power_t` while learning against changing conditions, like learning against an adversary who continuously changes the rules-of-the-game, would benefit from low (close to 0) `--power_t` so the learner can react quickly to these changing conditions. For many problems, 0.5, which is the default, seems to work best.
+
+This diagnostic shows the power on learning rate decay. The VW default is `0.5` and works well for most problems.
```
Comment (mikedropmedia) [inline _guides/getting_started.md]: @kumpera comments:
The change from "on" to "of" changes the meaning of the expression. It power in the mathematical sense and not metaphorical one.
The diagnostic output only shows what was used.

Gotcha

File: _guides/getting_started.md [hunk 7]
```diff
@@ -57,64 +86,140 @@ best constant = 0.500000
...
+### Learning rate decay
+
+The following VW output sample shows the initial time:
 
 ```
 initial_t = 0
 ```
-Learning rates should often decay over time, and this specifies the initial time. You can adjust with `--initial_t time`, although this is rarely necessary these days.
+
+This diagnostic shows the initial time duration for learning rate decay.
```
Comment (mikedropmedia) [inline _guides/getting_started.md]: @kumpera comments:
Diagnostic output only shows what was used.

Please confirm this is accurate, changed to "shows", deleted "or"

File: _guides/getting_started.md [hunk 8]
```diff
@@ -57,64 +86,140 @@ best constant = 0.500000
...
 ```
 initial_t = 0
 ```
-Learning rates should often decay over time, and this specifies the initial time. You can adjust with `--initial_t time`, although this is rarely necessary these days.
+
+This diagnostic shows the initial time duration for learning rate decay. 
+
+**Use** `--initial_t time` to adjust the initial time. 
+
+>**Note:** It is common to need to adjust the initial time in Vowpal Wabbit.
```
Comment (mikedropmedia) [inline _guides/getting_started.md]: @kumpera comments:
We're talking about options that the user might need to use with VW to achieve their goal.
Saying that it's not common doesn't tell anything about why.

Please confirm accuracy of statement now, "it IS common TO NEED to adjust `--initial_t time`...

**Review by mikedropmedia — COMMENTED** [review]
Addressed Rodrigo's edits, ready for review

**Review by mikedropmedia — COMMENTED** [review]
Ok, small fix now doc is ready for final review

File: _guides/getting_started.md [hunk 9]
```diff
@@ -10,31 +10,60 @@ body_class: tutorial
...
-# A Step by Step Introduction to Linear Regression
+# Getting Started with Linear Regression and Vowpal Wabbit
 
-## A first data-set
+This guide describes how to perform Linear Regression with Vowpal Wabbit (VW). It features an overview of a simple regression problem using a VW workflow tutorial with examples. You will find information about interacting with VW, including structuring input, understanding output, and learn important VW diagnostics. 
 
-Now, let's create a data-set. Suppose we want to predict whether a house will require a new roof in the next 10 years. We can create a training-set file, `house_dataset` with the following contents:
+## Getting started with Vowpal Wabbit
+
+To install VW—and for more information on building VW from source or using a package manager—see [Getting started](https://github.com/VowpalWabbit/vowpal_wabbit/wiki/Getting-started) guide.
```
Comment (kumpera) [inline _guides/getting_started.md]: We need to fix the links to point to the right place. @cheng-tan  could you help Mike use the proper relative link syntax here?

File: _guides/getting_started.md (hunk 2 above)
Comment (kumpera) [reply _guides/getting_started.md]: The Python link is wrong, people should use the install instructions to get to it.
I'm confused why we have a note about the CB tutorial here.

File: _guides/getting_started.md (hunk 3 above)
Comment (kumpera) [reply _guides/getting_started.md]: I was refering to the tag `second_house`. It's used later in the prediction output part.

File: _guides/getting_started.md [hunk 10]
```diff
@@ -57,64 +86,140 @@ best constant = 0.500000
...
+Next, VW shows you some diagnostic information in the header.
 
 ```
-average  since         example        example  current  current  current
-loss     last          counter         weight    label  predict features
-0.000000 0.000000            1            1.0   0.0000   0.0000        5
-0.666667 1.000000            2            3.0   1.0000   0.0000        5
+average since  example example current current current
+loss  last  counter weight label predict features
+0.000000 0.000000      1 1.0 0.0000 0.0000    5
```
Comment (kumpera) [inline _guides/getting_started.md]: Revert this change. The vertical alignment is intentional and without it you can't tell what value is associated with which column.

File: _guides/getting_started.md [hunk 11]
```diff
@@ -57,64 +86,140 @@ best constant = 0.500000
...
-- `current label` tells you the label of the second example.
-- `current predict` tells you the prediction (before training) on the current example.
-- `current features` tells you how many features the current example has. This is great for debugging, and you'll note that we have 5 features when you expect 4. This happens, because VW has a default constant feature which is always added in. Use the `--noconstant` command-line option to turn it off.
+-**`average loss`** computes the <a href="http://hunch.net/~jl/projects/prediction_bounds/progressive_validation/coltfinal.pdf" target="_blank">progressive validation</a> loss. Progressive validation loss deviates like a test set, and hence is a reliable indicator of success on the first pass over any data-set.
+-**`since last`** is the progressive validation loss since the last printout.
+-**`example counter`** indicates which example is printed out. In this case,  example `2`.
+-**`example weight`** is the sum of the importance weights of examples so far. In this case, it's `3.0`, because the second example has an importance weight of `2`.
```
Comment (kumpera) [inline _guides/getting_started.md]:
```suggestion
-**`example weight`** is the sum of the importance weights of examples seen so far. In this case, it's `3.0`, because the second example has an importance weight of `2`.
```

I can't say what does"examples so far" suggests without "seen"

File: _guides/getting_started.md [hunk 12]
```diff
@@ -57,64 +86,140 @@ best constant = 0.500000
...
+-**`current label`** is the label of the second example.
+-**`current predict`** is the prediction (before training) on the current example.
+-**`current features`** is the number of features in the current example. 
+
+The `current features` diagnostic is excellent for debugging. Note that we have five features when you expect four. This change happens because VW always adds a default _constant feature_.
+
+**Use** `--noconstant` the default constant feature off in VW.
```
Comment (kumpera) [inline _guides/getting_started.md]:
```suggestion
**Use** `--noconstant` to disable the default constant feature. 
```

File: _guides/getting_started.md [hunk 13]
```diff
@@ -57,64 +86,140 @@ best constant = 0.500000
...
+-**`current features`** is the number of features in the current example. 
+
+The `current features` diagnostic is excellent for debugging. Note that we have five features when you expect four. This change happens because VW always adds a default _constant feature_.
+
+**Use** `--noconstant` the default constant feature off in VW. 
 
-VW prints a new line with an exponential backoff. This is very handy, because often you can debug some problem before the learning algorithm finishes going through a data-set.
+## Vowpal Wabbit debugging diagnostics 
+
+Now, VW prints a new line with an exponential backoff. This information is useful for debugging problems before the learning algorithm finishes with a data-set:
```
Comment (kumpera) [inline _guides/getting_started.md]: This paragraph has no connection with the next line. Revert this edit.

File: _guides/getting_started.md (hunk 5 above)
Comment (kumpera) [inline _guides/getting_started.md]:
```suggestion
This diagnostic shows that the number of bits used from the hash function is 18 (more than enough for this example). Use `-b bits` to adjust the number of bits to use.
```

Btw, should we explain what we use those bits for and why?

File: _guides/getting_started.md [hunk 14]
```diff
@@ -43,10 +72,10 @@ power_t = 0.5
 using no cache
 Reading datafile = house_dataset
 num sources = 1
-average  since         example        example  current  current  current
-loss     last          counter         weight    label  predict features
-0.000000 0.000000            1            1.0   0.0000   0.0000        5
-0.666667 1.000000            2            3.0   1.0000   0.0000        5
+average since  example example current current current
```
Comment (kumpera) [inline _guides/getting_started.md]: We should keep the vertical spacing. The columns otherwise don't show aligned.

File: _guides/getting_started.md [hunk 15]
```diff
@@ -57,64 +86,140 @@ best constant = 0.500000
...
+The following VW output shows the learning rate:
 
 ```
 learning rate = 0.5
 ```
-The default learning rate is 0.5 which we found to be a good default with the current default update (`--normalized --invariant --adaptive`). If the data is noisy you'll need a larger data-set and/or multiple passes to predict well. On these larger data-sets, our learning rate will by default decay towards 0 as we run through examples. You can adjust the learning rate up or down with `-l rate`. A higher learning rate will make the model converge faster but a too high learning rate may over-fit and end-up worse on average.
+
+The default learning rate is `0.5` with current default update (`--normalized --invariant --adaptive`). 
+
+If the data is noisy, you need a larger data-set or multiple passes to predict well. For massive data-sets, the learning rate decays to `0` by default.
```
Comment (kumpera) [inline _guides/getting_started.md]:
```suggestion
If the data is noisy, you need a larger data-set or multiple passes to predict well. For massive data-sets, the learning rate decays towards `0` by default. 
```

Learning rate doesn't get to zero, but almost.

File: _guides/getting_started.md (hunk 7 above)
Comment (kumpera) [inline _guides/getting_started.md]:
```suggestion
This diagnostic shows the initial time for learning rate decay. 
```

File: _guides/getting_started.md (hunk 8 above)
Comment (kumpera) [reply _guides/getting_started.md]: This statement is in contradiction with the original. Setting is rarely needed.

Comment (kumpera) [inline _guides/getting_started.md]:
```suggestion
>**Note:** Changing this value is rarely needed with VW. 
```

File: _guides/getting_started.md [hunk 16]
```diff
@@ -57,64 +86,140 @@ best constant = 0.500000
...
 ```
-Learning rates should often decay over time, and this specifies the initial time. You can adjust with `--initial_t time`, although this is rarely necessary these days.
+
+This diagnostic shows the initial time duration for learning rate decay. 
+
+**Use** `--initial_t time` to adjust the initial time. 
+
+>**Note:** It is common to need to adjust the initial time in Vowpal Wabbit. 
+
+### Learning rate decay power
```
Comment (kumpera) [inline _guides/getting_started.md]: This is not about the power of the learning rate decay. This is about the mathematical power operator applied to the learning rate decay. It's power as in the phrase "3 to the power of 2 is equal to 9"

File: _guides/getting_started.md (hunk 6 above)
Comment (kumpera) [reply _guides/getting_started.md]: This needs to be changed in all previous parts that refers to power_t as well.

File: _guides/getting_started.md [hunk 17]
```diff
@@ -57,64 +86,140 @@ best constant = 0.500000
...
+
+This diagnostic shows the power on learning rate decay. The VW default is `0.5` and works well for most problems. 
+
+The `0.5` default is a minimax optimal choice for stationary data-sets where the fundamental relation between the input features and target label do not change over time and benefit from a high (close to 1.0) `--power_t` while learning against changing conditions.
```
Comment (kumpera) [inline _guides/getting_started.md]: 0.5 has nothing to do with stationary data-sets.


Move `0.5 is the default and a minimax optimal choice.` to the list of interesting values.

`A different way of stating this is: stationary data-sets where the fundamental relation between the input features and target label are not changing over time, should benefit from a high (close to 1.0) `--power_t` while learning against changing conditions, like learning against an adversary who continuously changes the rules-of-the-game, would benefit from low (close to 0) `--power_t` so the learner can react quickly to these changing conditions.` is an explanation of how to pick the value.

File: _guides/getting_started.md [hunk 18]
```diff
@@ -57,64 +86,140 @@ best constant = 0.500000
...
+### Cache files
+
+The following VW output shows that you are not using a cache file: 
 
 ```
 using no cache
 ```
-This says you are not using a cache. If you use multiple passes with `--passes`, you would need to also pass `-c` so VW can cache the data in a faster to handle format (passes > 1 should be much faster). By default, the cache file name will be the data-set file with `.cache` appended. In this case: `house_dataset.cache`. You may also override the default cache file name by passing: `--cache_file housing.cache`. A cache file can greatly speed up training when you run many experiments (with different options) on the same data-set even if each experiment is only a single pass. So if you want to experiment with the same data-set over and over, it is highly recommended to pass `-c` every time you train. If the cache exists and is newer than the data-set, it will be used, if it doesn't exist, it'll be created the first time `-c` is used.
+
+A cache file speeds up training when you run many experiments (with different options) on the same data-set—even if each experiment is only a single pass.
```
Comment (kumpera) [inline _guides/getting_started.md]: This wording is bad. It draws conclusions from the examples of where caching is helpful.
Something long this lines would be better.
```
A cache file contains your data-set in a faster to handle format.

A cache file can greatly speed up training if you're either using multiple passes or running multiple experiments on the same data-set (even with different options).
```

File: _guides/getting_started.md [hunk 19]
```diff
@@ -57,64 +86,140 @@ best constant = 0.500000
...
+
+A cache file speeds up training when you run many experiments (with different options) on the same data-set—even if each experiment is only a single pass. 
+
+The default cache file name is the data-set file name with `.cache` appended. For example, `house_dataset.cache`. 
+
+**Use** `--cache_file housing.cache` to override the default cache file name. 
+
+The first time you use `-c` you create a cache file. If the cache already exists and it is newer than the data-set, that is the default cache.  
+
+**Use** `-c` for multiple passes `--passes`, so VW caches the data in a faster format (passes > 1 should be much faster).
```
Comment (kumpera) [inline _guides/getting_started.md]: This breakdown of use/note is confusing. This is a recommendation for both cases.

File: _guides/getting_started.md [hunk 20]
```diff
@@ -57,64 +86,140 @@ best constant = 0.500000
...
+### Data sources 
+
+The following VW output shows the source of data: 
 
 ```
 Reading datafile = house_dataset
 ```
-There are many different ways to input data to VW. Here we're just using a simple text file and VW tells us the source of the data. Alternative sources include cache files (from previous VW runs), stdin, or a tcp socket.
+
+VW supports a variety of different input methods. This example uses a simple text file, and VW displays a diagnostic with the source of the data.
```
Comment (kumpera) [inline _guides/getting_started.md]: `Display a diagnostic` rubs me really wrong. VW displays text and some of it are diagnostics.

File: _guides/getting_started.md [hunk 21]
```diff
@@ -57,64 +86,140 @@ best constant = 0.500000
...
+### Number of data sources
+
+The following VW output shows the number of data sources:
 
 ```
 num sources = 1
 ```
-There is only one input file in our example. But you can specify multiple files.
 
+There is only one input file in this example. You can specify multiple files in VW.
```
Comment (kumpera) [inline _guides/getting_started.md]: This tutorial is only about VW, why keep repeating that's about VW all the time?

File: _guides/getting_started.md [hunk 22]
```diff
@@ -126,113 +231,155 @@ best constant = 0.500000
...
-0.184476 0.002182           23           31.0   1.0000   0.9975        5
-0.090774 0.000000           47           63.0   1.0000   1.0000        5
+average since  example example current current current
+loss  last  counter weight label predict features
+0.000000 0.000000      1 1.0 0.0000 0.0000    5
+0.666667 1.000000      2 3.0 1.0000 0.0000    5
+0.589385 0.531424      5 7.0 1.0000 0.2508    5
+0.378923 0.194769     11 15.0 1.0000 0.8308    5
+0.184476 0.002182     23 31.0 1.0000 0.9975    5
+0.090774 0.000000     47 63.0 1.0000 1.0000    5
```
Comment (kumpera) [inline _guides/getting_started.md]: Don't change formating of output blocks. It makes them unreadable.

File: _guides/getting_started.md [hunk 23]
```diff
@@ -126,113 +231,155 @@ best constant = 0.500000
...
 ```
 
-You'll notice that by example 47 (25 passes over 3 examples result in 75 examples), the `since last` column has dropped to 0, implying that by looking at the same (3 lines of) data 25 times we have reached a perfect predictor. This is unsurprising with 3 examples having 5 features each. The reason we have to add `--holdout_off` is that when running multiple-passes, VW automatically switches to 'over-fit avoidance' mode by holding-out 10% of the examples (the period "one in 10" can be changed using `--holdout_period period`) and evaluating performance on the held-out data instead of using the online-training progressive loss.
+Notice that the `since last` data dropped to `0` on line six, `example 47` (25 passes over 3 examples result in 75 examples). This result implies that, by looking at the same 3 lines of data 25 times, we reached a _perfect predictor_. Not surprising, given that with three examples and five features for each example.
```
Comment (kumpera) [inline _guides/getting_started.md]: `Notice that the `since last` data dropped to `0` on line six, `example 47` (25 passes over 3 examples result in 75 examples)` Makes no sense.

File: _guides/getting_started.md [hunk 24]
```diff
@@ -126,113 +231,155 @@ best constant = 0.500000
...
+Notice that the `since last` data dropped to `0` on line six, `example 47` (25 passes over 3 examples result in 75 examples). This result implies that, by looking at the same 3 lines of data 25 times, we reached a _perfect predictor_. Not surprising, given that with three examples and five features for each example. 
 
-## Saving your model (a.k.a. regressor) into a file
+>**Note:** You have to add `--holdout_off` when running multiple-passes. VW automatically switches to _over-fit avoidance_ mode by holding 10% of the examples and evaluating performance on the data being held instead of using online training progressive loss algorithms.
```
Comment (kumpera) [inline _guides/getting_started.md]: This is wrong. Only the example needed --holdoud_off and not when using multiple passes.

File: _guides/getting_started.md [hunk 25]
```diff
@@ -126,113 +231,155 @@ best constant = 0.500000
...
 
-By default VW learns the weights of the features and keeps them in an in memory vector. If you want to save the final regressor weights into a file, add `-f filename`:
+**Use** `--holdout_period period` to change the one in 10 default holdout in VW. 
+
+## Saving regressor model files
```
Comment (kumpera) [inline _guides/getting_started.md]:
```suggestion
## Saving your model into a file
```

File: _guides/getting_started.md [hunk 26]
```diff
@@ -126,113 +231,155 @@ best constant = 0.500000
...
+
+Vowpal Wabbit learns the weights of the features and keeps them in an _in-memory vector_ by default. 
+
+**Add** `-f filename` to create and save the final regressor file. For example:
```
Comment (kumpera) [inline _guides/getting_started.md]:
```suggestion
**Add** `-f filename` to save the final regressor weights to a file. For example:
```

File: _guides/getting_started.md [hunk 27]
```diff
@@ -126,113 +231,155 @@ best constant = 0.500000
...
 
 ```sh
 vw house_dataset -l 10 -c --passes 25 --holdout_off -f house.model
 ```
 
-## Getting predictions
+## Make predictions with Vowpal Wabbit 
 
-We want to make predictions of course, this can be done by supplying the `-p filename` option. Stdout can be used as below:
+You can make predictions in VW by supplying the `-p filename`. For example, using filename `stdout`:
```
Comment (kumpera) [inline _guides/getting_started.md]:
```suggestion
You can make predictions in VW by supplying the `-p filename`. For example, to output them to standard out
```

File: _guides/getting_started.md [hunk 28]
```diff
@@ -126,113 +231,155 @@ best constant = 0.500000
...
 ```
 
-- The 1st output line `0.000000` is for the 1st example which has an empty tag.
-- The 2nd output `0.000000 second_house` is for the 2nd example. You'll notice the tag appears here. This is the primary use of the tag: mapping predictions to the examples they belong to.
-- The 3rd output `1.000000 third_house` is for the 3rd example. Clearly, some learning happened, because the prediction is now `1.000000` while the initial prediction was set to `0.5`.
+The first line, `0.000000` refers to the first example, which has an empty tag.
+The second line, `0.000000 second_house` refers to the second example. Notice that the tag appears here. The primary use of the tag is mapping predictions to the corresponding examples.
+The third output, `1.000000 third_house` refers to the third example. The initial prediction was set to `0.5`, and the prediction is now `1.000000`. This result means that _some_ learning occurred. 
+
+In the last example, the model was built in memory incrementally, as VW analyzed the examples. In other words, you predicted _while you learned_.
```
Comment (kumpera) [inline _guides/getting_started.md]: analyzed has no meaning here.

Comment (kumpera) [inline _guides/getting_started.md]:
```suggestion
In the last example, VW predicted while it learned. The model was being built in memory incrementally, as VW went over the examples.
```

File: _guides/getting_started.md [hunk 29]
```diff
@@ -126,113 +231,155 @@ best constant = 0.500000
...
 
-Note that in this last example, we predicted _while we learned_. The model was being incrementally built in memory as VW went over the examples.
+It is more common to learn first, then save the model file. Then, you make predictions using that saved model.
```
Comment (kumpera) [inline _guides/getting_started.md]:
```suggestion
It is more common to learn first, then save the model to a file. Then, you make predictions using that saved model.
```

File: _guides/getting_started.md [hunk 30]
```diff
@@ -126,113 +231,155 @@ best constant = 0.500000
...
+Notice that the `since last` data dropped to `0` on line six, `example 47` (25 passes over 3 examples result in 75 examples). This result implies that, by looking at the same 3 lines of data 25 times, we reached a _perfect predictor_. Not surprising, given that with three examples and five features for each example. 
 
-## Saving your model (a.k.a. regressor) into a file
+>**Note:** You have to add `--holdout_off` when running multiple-passes. VW automatically switches to _over-fit avoidance_ mode by holding 10% of the examples and evaluating performance on the data being held instead of using online training progressive loss algorithms.
 
-By default VW learns the weights of the features and keeps them in an in memory vector. If you want to save the final regressor weights into a file, add `-f filename`:
+**Use** `--holdout_period period` to change the one in 10 default holdout in VW.
```
Comment (kumpera) [inline _guides/getting_started.md]: the "1 in 10" period.

File: _guides/getting_started.md [hunk 31]
```diff
@@ -126,113 +231,155 @@ best constant = 0.500000
...
+The output for this command is:
+
 ```
 0.000000
 1.000000 second_house
 0.000000 third_house
 ```
 
-Obviously the results are different this time, because in the first prediction example, we learned as we went, and made only one pass over the data, whereas in the 2nd example we first loaded an over-fitted (25 pass) model and used our data-set `house_dataset` with `-t` (testing only mode). In real prediction settings, one should use a different data-set for testing vs training.
+The results are different for this example because the first prediction example made only one pass over the data, in other words, learning as it worked.
```
Comment (kumpera) [inline _guides/getting_started.md]: This words is confusing and wrong.

The example reference is for the invocation of vw right before this one.

File: _guides/getting_started.md [hunk 32]
```diff
@@ -126,113 +231,155 @@ best constant = 0.500000
...
 
-Obviously the results are different this time, because in the first prediction example, we learned as we went, and made only one pass over the data, whereas in the 2nd example we first loaded an over-fitted (25 pass) model and used our data-set `house_dataset` with `-t` (testing only mode). In real prediction settings, one should use a different data-set for testing vs training.
+The results are different for this example because the first prediction example made only one pass over the data, in other words, learning as it worked. 
+
+For this example, you loaded the over-fitted model (25 passes) and used the `house_dataset` with `-t` (or _testing only_ mode). 
+
+>**Note:** Always use a different data-set for testing vs. training for real prediction settings.
 
-## Auditing
+## Auditing machine learning applications with Vowpal Wabbit
```
Comment (kumpera) [inline _guides/getting_started.md]: No, we're not auditing applications.

File: _guides/getting_started.md [hunk 33]
```diff
@@ -126,113 +231,155 @@ best constant = 0.500000
...
+  price:229902:0.53:0.882655@0.2592 age:165201:0.87:0.453833@0.98 sqft:162853:0.32:1.05905@0.18 Constant:116060:1:0.15882@8 1924:1924:1:0@0
 ```
 
-Every example uses two lines. The first line has the prediction, and the second line has one entry per feature. Looking at the first feature, we see:
+Every example uses two lines:
+
+The first line is the prediction.
+The second line shows one entry per feature. 
+
+VW has an advanced_namespaces_ feature that allows you to group features and operate them on-the-fly. Namespace options include the following:
```
Comment (kumpera) [inline _guides/getting_started.md]:
```suggestion
VW has an advanced _namespaces_ feature that allows you to group features and operate them on-the-fly. Namespace options include the following: 
```

File: _guides/getting_started.md [hunk 34]
```diff
@@ -126,113 +231,155 @@ best constant = 0.500000
...
 ```
 
-- `price` is the original feature name. If you use a namespace, it appears before `^` (i.e. `Namespace^Feature`). Namespaces are an advanced feature which allows you to group features and operate them on-the-fly, in the core of VW with the options: `-q XY` (cross a pair of namespaces), `--cubic XYZ` (cross 3 namespaces), `--lrq XYn` (low-rank quadratic interactions), and `--ignore X` (skip all features belonging to a namespace).
-- `229902` is the index of the feature, computed by a hash function on the feature name.
-- `0.23` is the value of the feature.
-- `0` is the value of the feature's weight.
-- `@0.25` represents the sum of gradients squared for that feature when you are using per-feature adaptive learning rates.
+For the first feature listed, **`price`** is the original feature name. If you use a namespace, it appears before `^`. For example: 
+
+`Namespace^Feature`
```
Comment (kumpera) [inline _guides/getting_started.md:359]: This looks really confusing as it breaks the bullet point description of the line

File: _guides/getting_started.md [hunk 35]
```diff
@@ -126,113 +231,155 @@ best constant = 0.500000
...
-- `@0.25` represents the sum of gradients squared for that feature when you are using per-feature adaptive learning rates.
+For the first feature listed, **`price`** is the original feature name. If you use a namespace, it appears before `^`. For example: 
+
+`Namespace^Feature` 
+- ** `229902`** is the index of the feature, computed by a hash function on the feature name.
+- **`0.23`** is the value of the feature.
+- **`0`** is the value of the feature's weight.
+- **`@0.25**` is the sum of gradients squared for that feature (when you use _per-feature adaptive learning rates_).
+
+Notice that the feature `2006` uses the index 2006. Like other ML tools, you can use _hashes_ or _pre-computed indices_ for features in VW.
```
Comment (kumpera) [inline _guides/getting_started.md]: This changes the meaning of the phrase. Only pre-computed indices are common. hashes are not

**Review by kumpera — CHANGES_REQUESTED** [review]

File: _guides/getting_started.md (hunk 5 above)
Comment (mikedropmedia) [reply _guides/getting_started.md]: Yes, in an equal amount of space if possible? 

Probably doesn't warrant it's own section if that's what it would take.

File: _guides/getting_started.md (hunk 8 above)
Comment (mikedropmedia) [reply _guides/getting_started.md]: Changing to active voice: "It is rare to change this value in VW. 

Do you think it warrants a "why" here? Or explain an option where it is necessary to change the initial time?

File: _guides/getting_started.md (hunk 16 above)
Comment (mikedropmedia) [reply _guides/getting_started.md]: So change section H3 title to "Power Operator?" We need an accurate H3 here for consistency and flow.

File: _guides/getting_started.md (hunk 17 above)
Comment (mikedropmedia) [reply _guides/getting_started.md]: See updated

File: _guides/getting_started.md (hunk 18 above)
Comment (mikedropmedia) [reply _guides/getting_started.md]: Updated

File: _guides/getting_started.md (hunk 20 above)
Comment (mikedropmedia) [reply _guides/getting_started.md]: Deleted

File: _guides/getting_started.md (hunk 21 above)
Comment (mikedropmedia) [reply _guides/getting_started.md]: Good point

File: _guides/getting_started.md (hunk 10 above)
Comment (mikedropmedia) [reply _guides/getting_started.md]: Done

File: _guides/getting_started.md (hunk 13 above)
Comment (mikedropmedia) [reply _guides/getting_started.md]: We need something before the output example under the H2. Please add some copy to introduce or transition to the output example.

File: _guides/getting_started.md (hunk 22 above)
Comment (mikedropmedia) [reply _guides/getting_started.md]: This, and the others, was a formatting error when we copy/pasted from my backup file after losing commits.

File: _guides/getting_started.md (hunk 14 above)
Comment (mikedropmedia) [reply _guides/getting_started.md]: Done

File: _guides/getting_started.md (hunk 23 above)
Comment (mikedropmedia) [reply _guides/getting_started.md]: "line six" referenced the table data, I dropped that. The rest is an effort to hold the reader's hand when you ref table data and call attention to previous examples/output.

File: _guides/getting_started.md (hunk 19 above)
Comment (mikedropmedia) [reply _guides/getting_started.md]: See updated

File: _guides/getting_started.md (hunk 28 above)
Comment (mikedropmedia) [reply _guides/getting_started.md]: Is there a better, more accurate term to replace "went over?"

File: _guides/getting_started.md (hunk 34 above)
Comment (mikedropmedia) [reply _guides/getting_started.md:359]: Yes, it does, updated and rearranged. Please check for accuracy.

File: _guides/getting_started.md (hunk 35 above)
Comment (mikedropmedia) [reply _guides/getting_started.md]: Got it, reverted to emphasize indices is common, hashes more unique.

**Review by mikedropmedia — COMMENTED** [review]
Addressed Rodrigo's comments and concerns. 

Fixed copy/paste errors from backup doc.

File: _guides/getting_started.md [hunk 36]
```diff
@@ -10,31 +10,60 @@ body_class: tutorial
...
 
-## A first data-set
+This guide describes how to perform Linear Regression with Vowpal Wabbit (VW). It features an overview of a simple regression problem using a VW workflow tutorial with examples. You will find information about interacting with VW, including structuring input, understanding output, and learn important VW diagnostics. 
 
-Now, let's create a data-set. Suppose we want to predict whether a house will require a new roof in the next 10 years. We can create a training-set file, `house_dataset` with the following contents:
+## Getting started with Vowpal Wabbit
+
+To install VW—and for more information on building VW from source or using a package manager—see [Getting started](https://github.com/VowpalWabbit/vowpal_wabbit/wiki/Getting-started) guide. 
+
+>**Note:** This tutorial uses [Vowpal Wabbit Python package](https://github.com/VowpalWabbit/vowpal_wabbit/tree/master/python). Additional binary packages are available for select platforms. See [Getting started](https://cheng-tan.github.io/vowpalwabbit.github.io/) module on the homepage for more information.
```
Comment (cheng-tan) [inline _guides/getting_started.md]:
```suggestion
>**Note:** This tutorial uses [Vowpal Wabbit Python package](https://github.com/VowpalWabbit/vowpal_wabbit/tree/master/python). Additional binary packages are available for select platforms. See [Getting started]({{ site.baseurl }}/#get_started) module on the homepage for more information.
```

File: _guides/getting_started.md (hunk 9 above)
Comment (cheng-tan) [reply _guides/getting_started.md]: this links needs adding some code. On line 13, adding in 
`{% assign getting_started = site.guides | where: "module_id","getting_started" | first %}`

Then on line 19 replace the URL to `({{ site.baseurl }}{{getting_started.url}})`
